File: 2 Revalida en C++/5 Funciones/13.cpp
/**#funcion para en contrar el area bajo una curva
f(x)=x^2senx f(a)=0 f(b)=3.14159**/
#include <bits/stdc++.h>
using namespace std;
float trap(float a,float b,float n)
{
    float h,x,t;
    h=(b-a)/n;
	float suma=0;
	for (int i=1;i<=n-1;i++)
		{
		    x=a+i*h;
            suma=suma+(pow(x,2)*(sin(x)));
		}
	t=0.5*h*(0+3.1415+2*suma);
	return t;
}
int main()
{
    int a,b,n;
    cout<<"ingrese a:";
    cin>>a;
    cout<<"ingrese b:";
    cin>>b;
    cout<<"ingrese n:";
    cin>>n;
    cout<<"el resultado es ="<<trap(a,b,n);
    return 0;
}

File: 1 Revalida en python/5 Funciones/13.py
#funcion para en contrar el area bajo una curva
#f(x)=x^2senx f(a)=0 f(b)=3.14159
import math
def trap(a,b,n):
	h=(b-a)/n
	suma=0
	i=1
	for i in (1,(n-1)):
		x=a+i*h
		suma =suma+((x*x)*(math.sin(x)))
	t=0.5*h*(0+3.1415+2*suma)
	return t
a=float(input("ingrese un numero a:"))
b=float(input("ingrese un numero b:"))
n=float(input("ingrese un numero n:"))
print trap(a,b,n)